RealmTasks Android/app/src/main/java/io/realm/realmtasks/TaskActivity.java

```diff
@@ -155,20 +155,17 @@ public void onMoved(RecyclerView recyclerView, ItemViewHolder from, ItemViewHold
             final int fromPosition = from.getAdapterPosition();
             final int toPosition = to.getAdapterPosition();
             adapter.onItemMoved(fromPosition, toPosition);
-            adapter.notifyItemMoved(fromPosition, toPosition);
         }
 
         @Override
         public void onCompleted(ItemViewHolder viewHolder) {
             adapter.onItemCompleted(viewHolder.getAdapterPosition());
-            adapter.notifyDataSetChanged();
```

Comment:
the colour change should be handled by the fine grained notification as well, otherwise exception will be thrown. In this case, the whole screen needs to be refreshed to show different colours (I assume?) which makes fine grain notification useless. I suggest to change the design to demo something more important -- fine grained changes.